spec/tbd_tests_spec.rb

```diff
@@ -1059,13 +1057,93 @@
     expect(json).to have_key(:surfaces)
     io       = json[:io      ]
     surfaces = json[:surfaces]
-    expect(TBD.status).to be_zero
-    expect(TBD.logs).to be_empty
+    expect(TBD.error?).to be true
+    expect(TBD.logs.size).to eq(1)
+    expect(TBD.logs.first[:message]).to include("Exiting - KIVA objects in ")
+    expect(surfaces).to be_a(Hash)
+    expect(surfaces.size).to eq(56)
+    expect(io).to be_a(Hash)
+    expect(io).to have_key(:edges)
+
+    # 105x edges (-1x than the usual 106x for the seb.osm). The edge linking
+    # "Open area 1 Floor" to "Openarea 1 Wall 5" used to be of type :grade. As
+    # both slab and wall are now ground-facing, TBD ignores the edge altogether.
+    expect(io[:edges].size).to eq(105)
+    expect(model.foundationKivaSettings).to be_empty
+    expect(model.getSurfacePropertyExposedFoundationPerimeters.size).to eq(1)
+    expect(model.getFoundationKivas.size).to eq(4)
+
+    # TBD derates (above-grade) surfaces as usual. TBD is certainly 'aware' of
+    # the "Foundation"-facing slab and wall (and their shared edge), yet exits
+    # the KIVA generation step. As the warning message suggests, TBD safely
+    # exits when the OpenStudio model already holds KIVA objects.
+    surfaces.values.each { |surface| expect(surface).to_not have_key(:kiva) }
+
+    # As with the previously altered "files/osms/out/seb_KIVA.osm", OpenStudio
+    # can forward-translate and run an EnergyPlus simulation without warnings or
+    # errors. As "Openarea 1 Wall 5" is now a "Foundation"-facing wall, the
+    # exposed foundation perimeter length (set previously) is now invalid. Yet
+    # there are no internal checks in OpenStudio and/or EnergyPlus to ensure
+    # perimeter length consistency, WHEN exposed + foundation perimeter
+    # lengths < total slab perimeter lengths. Simulation runs without a glitch;
+    # simulation results would be 'off'.
```

Comment:
Additional comments to explain what happens when TBD purges existing KIVA objects. Users may not be completely out of the woods, but EnergyPlus runs without related warnings or errors.